agrirouter-sdk-dotnet-standard-impl/Service/Messaging/FetchMessageService.cs

```diff
@@ -29,10 +30,33 @@ public FetchMessageService(HttpClient httpClient)
         }
 
         /// <summary>
-        ///     Fetch messages from the inbox using the given onboarding response.
+        /// Fetch messages from the inbox using the given onboard response and an implementation of a cancellation token to perform polling over a certain amount of time / tries.
         /// </summary>
-        /// <param name="onboardResponse">All the messages that are in the inbox.</param>
-        /// <returns>-</returns>
+        /// <param name="onboardResponse">Onboard response to connect to the agrirouter.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the polling process.</param>
+        /// <returns>All the messages that are in the inbox.</returns>
+        /// <exception cref="CouldNotFetchMessagesException">Will be thrown if the messages can not be fetched.</exception>
+        public List<MessageResponse> Fetch(OnboardResponse onboardResponse, ICancellationToken cancellationToken)
+        {
+            var totalMessageResponses = new List<MessageResponse>();
+            while (cancellationToken.IsNotCancelled())
+            {
+                cancellationToken.WaitBeforeNextStep();
```

Comment:
Shouldn't this be called after the Fetch and even after the cancelationToken.NextStep()?
Otherwise, how do I call the function with one try and no delay?

Reply:
After `Fetch` is fine, but it should be before performing `NextStep`. Otherwise, it has to be renamed to `WaitAfterNextStep`.